libraries/standard/http/cbmc/sources/http_cbmc_state.c

```diff
@@ -0,0 +1,38 @@
+#include <stdlib.h>
+
+#include "http_cbmc_state.h"
+
+extern _Bool nondet_bool();
+
+void * safeMalloc( size_t wantedSize )
+{
+    return nondet_bool() ? malloc( wantedSize ) : NULL;
+}
+
+HTTPRequestHeaders_t * allocateHttpRequestHeaders()
+{
+    HTTPRequestHeaders_t * pRequestHeaders = NULL;
+
+    pRequestHeaders = safeMalloc( sizeof( HTTPRequestHeaders_t ) );
+
+    if( pRequestHeaders )
+    {
+        pRequestHeaders->pBuffer = safeMalloc( pRequestHeaders->bufferLen );
+    }
+
+    return pRequestHeaders;
+}
+
+int isValidHttpRequestHeaders( const HTTPRequestHeaders_t * pRequestHeaders )
+{
+    int validBuffer = 1;
```

Comment:
Probably want to check that pRequestHeaders is nonnull.  Kind of up to you to decide whether a null pointer can be considered valid, but you want to decide before you dereference it below.